pandas/_libs/tslib.pyx

```diff
@@ -598,7 +597,9 @@ cpdef array_to_datetime(
         #    (with individual dateutil.tzoffsets) are returned
         is_same_offsets = len(out_tzoffset_vals) == 1
         if not is_same_offsets:
-            return _array_to_datetime_object(values, errors, dayfirst, yearfirst)
+            raise ValueError(
```

Comment:
I think the comments above should also be revised cc @MarcoGorelli

Reply:
yup, thanks

Reply:
sorry, I missed this comment. I replaced it with
```
# 2) If the offsets are different, then do not force the parsing
#    and raise a ValueError: "cannot parse datetimes with
#    mixed time zones unless `utc=True`" instead
```